src/ncp/ncp_openthread.cpp

```diff
@@ -245,7 +247,7 @@ bool ControllerOpenThread::IsAutoAttachEnabled(void)
     const char *val = getenv("OTBR_NO_AUTO_ATTACH");
 
     // Auto Thread attaching is enabled if OTBR_NO_AUTO_ATTACH is unset, empty or "0"
-    return (val == nullptr || !strcmp(val, "") || !strcmp(val, "0"));
+    return (val == nullptr || !strcmp(val, "") || !strcmp(val, "0")) && mEnableAutoAttach;
```

Comment:
We can remove `OTBR_NO_AUTO_ATTACH` logic to always use `--auto-attach` ?

We can add `--auto-attach 0` arguments when exec to spawn the new process.

Reply:
Done.